inst/tests/tests.Rraw

```diff
@@ -18278,3 +18278,13 @@ test(2245.1, dt[1], data.table(foo = 1L, bar = 4L)) # Single index should be int
 test(2245.2, copy(dt[1]), data.table(foo = 1:3)) # Revert to data.frame syntax: Interpret index as column
 rm(.datatable.aware)
 test(2245.3, dt[1], data.table(foo = 1L, bar = 4L)) # Default in this environment should be datatable-aware
+
+# 5392 split(x,f) does not work with formula f
+dt = data.table(x=1:4, y=factor(letters[1:2]))
+test(2246.1, split(dt, ~y), list(a=data.table(x=c(1L,3L), y=factor("a", levels=c("a", "b"))), b=data.table(x=c(2L, 4L), y=factor("b", levels=c("a", "b")))))
```

Comment:
probably clearer to make obvious that `~y` is intended as a shorthand for `dt$y`

```suggestion
test(2246.1, split(dt, ~y), split(dt, dt$y))
```